lrpar/src/lib/parser.rs

```diff
@@ -331,25 +347,29 @@ where
                 Action::Reduce(pidx) => {
                     let ridx = self.grm.prod_to_rule(pidx);
                     let pop_idx = pstack.len() - self.grm.prod(pidx).len();
-                    if let Some(ref mut tstack_uw) = *tstack {
-                        let nodes = tstack_uw
-                            .drain(pop_idx - 1..)
-                            .collect::<Vec<Node<StorageT>>>();
-                        tstack_uw.push(Node::Nonterm { ridx, nodes });
+                    if let Some(ref mut astack_uw) = *astack {
+                        if let Some(f) = self.actions[usize::from(pidx)] {
+                            let v = AStackType::ActionType(f(
+                                ridx,
+                                "", // XXX,
```

Comment:
What does the `XXX` stand for?

Reply:
Good question. I should have explained this a bit better in the commit message (it's hinted at in the last paragraph): these are points where the API wanted us to pass around the entire user input, which wasn't practical. These XXXs disappear in https://github.com/softdevteam/grmtools/pull/56/commits/c1e352dd0ba8318772e79083971a6b5a6005dfa7

Reply:
I've just read the final commit where it became clear these were placeholders. Probably not worth removing them then.